zone/attack.cpp

```diff
@@ -1042,6 +1042,20 @@ void Mob::MeleeMitigation(Mob *attacker, DamageHitInfo &hit, ExtraAttackOptions
 
 	auto roll = RollD20(hit.offense, mitigation);
 
+	// Add bonus to roll if level difference is sufficient
+	signed lvldiff = attacker->GetLevel() - this->GetLevel();
+	if (RuleI(Combat, LevelDifferenceRollCheck) >= 0 && lvldiff > RuleI(Combat, LevelDifferenceRollCheck)) {
+		roll += RuleR(Combat, LevelDifferenceRollBonus);
+		if (roll > 2.0) {
+			roll = 2.0;
+		}
+	} else if (RuleI(Combat, LevelDifferenceRollCheck) >= 0 && lvldiff < (-RuleI(Combat, LevelDifferenceRollCheck))) {
+		roll -= RuleR(Combat, LevelDifferenceRollBonus);
+		if (roll < 0.1) {
+			roll = 0.1;
+		}
+	}
```

Comment:
In this case since the types that we're using for the rule and a variable are consistent, it'd save code duplication to use a variable for all of these.
```cpp
int level_diff_roll_check = RuleI(Combat, LevelDifferenceRollCheck);
```